### PR TITLE
Show cardView with the forecast

### PR DESCRIPTION
**Pull request by jnicolaslaurenti**
In this PR, the app show a recyclerView with the forecast information

**_Motorola Z Play:_**

ezgif com-gif-maker (4)

### CODE REVIEW

File: app/src/main/java/com/onboarding/wheatherapp/WeatherAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    override fun getItemCount() = days.size
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) =
+        WeatherViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.item_weather_day, parent, false))
+
+    companion object {
+        private const val URL = "http://openweathermap.org/img/w/"
+        private const val FORMAT = ".png"
+        private const val UNIT = "°C"
```
Comment (SartiMau) [inline app/src/main/java/com/onboarding/wheatherapp/WeatherAdapter.kt]: put this on the string res

File: app/src/main/java/com/onboarding/wheatherapp/data/services/WeatherService.kt [hunk 2]
```diff
@@ -26,5 +29,6 @@ class WeatherService {
     companion object {
         private const val APPID = "90c4a6b89d1036c5fb537a1a3582e818"
```
Comment (SartiMau) [inline app/src/main/java/com/onboarding/wheatherapp/data/services/WeatherService.kt]: put this APPID in a variable on gradle pls

File: app/src/main/java/com/onboarding/wheatherapp/mvp/view/ActivityView.kt [hunk 3]
```diff
@@ -1,11 +1,14 @@
...
+import android.content.Context
 import java.lang.ref.WeakReference
 
 open class ActivityView(activity: Activity) {
     private val activityRef: WeakReference<Activity> = WeakReference(activity)
-
     val activity: Activity?
         get() = activityRef.get()
+    val context: Context?
+        get() = activity
```
Comment (SartiMau) [inline app/src/main/java/com/onboarding/wheatherapp/mvp/view/ActivityView.kt:12]: you can put here activity.context

File: app/src/main/res/layout/activity_main.xml [hunk 4]
```diff
@@ -4,16 +4,38 @@
     xmlns:tools="http://schemas.android.com/tools"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
+    android:background="@color/background_activity"
     tools:context=".MainActivity">
 
+    <ImageView
```
Comment (SartiMau) [inline app/src/main/res/layout/activity_main.xml:10]: pls format the code properly

File: app/src/main/res/layout/item_weather_day.xml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+            android:id="@+id/TempMax"
+            style="@style/tempMinMax_style"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            tool:text="@string/temp_max"
+            app:layout_constraintTop_toTopOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"/>
+    </androidx.constraintlayout.widget.ConstraintLayout>
+
+</androidx.cardview.widget.CardView>
```
Comment (SartiMau) [inline app/src/main/res/layout/item_weather_day.xml]: add an empty line at the end pls

File: app/src/main/java/com/onboarding/wheatherapp/data/mapper/WeatherMapper.kt [hunk 6]
```diff
@@ -2,22 +2,33 @@ package com.onboarding.wheatherapp.data.mapper
...
 import com.onboarding.wheatherapp.data.entity.Data
 import com.onboarding.wheatherapp.data.entity.Main
-import com.onboarding.wheatherapp.data.entity.Weather
+import com.onboarding.wheatherapp.data.entity.WeatherDescription
 import com.onboarding.wheatherapp.data.services.response.DataResponse
 import com.onboarding.wheatherapp.data.services.response.MainResponse
-import com.onboarding.wheatherapp.data.services.response.WeatherResponse
+import com.onboarding.wheatherapp.data.services.response.WeatherDescriptionResponse
+import java.text.SimpleDateFormat
+import java.util.*
```
Comment (BQuinteros) [inline app/src/main/java/com/onboarding/wheatherapp/data/mapper/WeatherMapper.kt]: pls remove this wildcard

File: app/src/main/res/values/dimens.xml [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    <dimen name="image_width">70dp</dimen>
+    <dimen name="image_height">70dp</dimen>
+    <dimen name="image_padding_left">14dp</dimen>
+    <dimen name="image_padding_right">14dp</dimen>
+    <dimen name="temp_now_margin_bottom">2dp</dimen>
+    <dimen name="temp_now_text_size">24sp</dimen>
+    <dimen name="temp_min_max_margin_end">6dp</dimen>
+    <dimen name="temp_min_max_margin_right">6dp</dimen>
+    <dimen name="temp_min_max_text_size">20sp</dimen>
+    <dimen name="activity_main_text_view_height">70sp</dimen>
```
Comment (SartiMau) [inline app/src/main/res/values/dimens.xml]: activity_main_text_view_height, activity_main_image_view_width and activity_main_image_view_height should be in dp not sp

File: app/src/main/java/com/onboarding/wheatherapp/WeatherAdapter.kt [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+        fun bind(item: Data) {
+            val requestOptions = RequestOptions().placeholder(R.drawable.icon_default).error(R.drawable.icon_default)
+            binding.apply {
+                Glide.with(itemView.context)
+                    .applyDefaultRequestOptions(requestOptions)
+                    .load("$URL${item.weather.first().iconWeather}$FORMAT")
+                    .into(weatherIcon)
+                item.main.apply {
+                    date.text = item.date.toString()
+                    TempMin.text = tempMin + UNIT
```
Comment (BQuinteros) [inline app/src/main/java/com/onboarding/wheatherapp/WeatherAdapter.kt]: we can do here something like this "$tempMin$UNIT"

File: app/src/main/res/values/styles.xml [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+    <style name="tempMinMax_style">
+        <item name="android:layout_margin">8dp</item>
+        <item name="android:layout_marginEnd">6dp</item>
+        <item name="android:layout_marginRight">6dp</item>
+        <item name="android:gravity">bottom</item>
+        <item name="android:textStyle">italic</item>
+        <item name="android:textSize">20sp</item>
+    </style>
+
+</resources>
```
Comment (SartiMau) [inline app/src/main/res/values/styles.xml]: same

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/onboarding/wheatherapp/data/services/WeatherService.kt (hunk 2 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/wheatherapp/data/services/WeatherService.kt]: Solved!

File: app/src/main/res/layout/activity_main.xml (hunk 4 above)
Comment (jnicolaslaurenti) [reply app/src/main/res/layout/activity_main.xml:10]: Solved!

File: app/src/main/res/layout/item_weather_day.xml (hunk 5 above)
Comment (jnicolaslaurenti) [reply app/src/main/res/layout/item_weather_day.xml]: Solved!

File: app/src/main/res/values/dimens.xml (hunk 7 above)
Comment (jnicolaslaurenti) [reply app/src/main/res/values/dimens.xml]: Solved!

File: app/src/main/java/com/onboarding/wheatherapp/WeatherAdapter.kt (hunk 8 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/wheatherapp/WeatherAdapter.kt]: Solved!

File: app/src/main/java/com/onboarding/wheatherapp/data/mapper/WeatherMapper.kt (hunk 6 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/wheatherapp/data/mapper/WeatherMapper.kt]: Solved!

File: app/src/main/java/com/onboarding/wheatherapp/WeatherAdapter.kt (hunk 1 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/wheatherapp/WeatherAdapter.kt]: Solved!

File: app/src/main/java/com/onboarding/wheatherapp/mvp/view/ActivityView.kt (hunk 3 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/wheatherapp/mvp/view/ActivityView.kt:12]: According to what was discussed in the call, I will not modify it

File: app/src/main/res/values/styles.xml (hunk 9 above)
Comment (jnicolaslaurenti) [reply app/src/main/res/values/styles.xml]: Solved!

File: app/src/main/res/values/styles.xml [hunk 10]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+
+    <style name="temp_now_style">
+        <item name="android:layout_marginBottom">@dimen/temp_now_margin_bottom</item>
+        <item name="android:textColor">@color/black</item>
+        <item name="android:textSize">24sp</item>
```
Comment (SartiMau) [inline app/src/main/res/values/styles.xml]: use dimens pls

File: app/src/main/res/values/styles.xml [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+        <item name="android:textStyle">italic</item>
+    </style>
+
+    <style name="tempMinMax_style">
```
Comment (SartiMau) [inline app/src/main/res/values/styles.xml:11]: same here

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/onboarding/wheatherapp/data/mapper/WeatherMapper.kt [hunk 12]
```diff
@@ -2,22 +2,41 @@ package com.onboarding.wheatherapp.data.mapper
...
 
 class WeatherMapper {
 
     private fun transformMain(main: MainResponse) = Main(main.temp, main.tempMax, main.tempMin)
 
-    private fun transformList(list: ArrayList<DataResponse>): ArrayList<Data> {
+    private val apiFormat = SimpleDateFormat("yyyy-MM-dd HH:mm:ss", Locale.ENGLISH)
+    private val appFormat = SimpleDateFormat("EEEE, MM/dd", Locale.ENGLISH)
+
+    private fun transformWeatherDescription(weather: ArrayList<WeatherDescriptionResponse>): ArrayList<WeatherDescription> {
```
Comment (mauroLonghi) [inline app/src/main/java/com/onboarding/wheatherapp/data/mapper/WeatherMapper.kt]: why you are using a ArrayList ? please use a List or a MutableList in kotlin

File: app/src/main/res/values/styles.xml (hunk 10 above)
Comment (jnicolaslaurenti) [reply app/src/main/res/values/styles.xml]: Solved!

File: app/src/main/res/values/styles.xml (hunk 11 above)
Comment (jnicolaslaurenti) [reply app/src/main/res/values/styles.xml:11]: Solved!

**Review by SartiMau — APPROVED** [review]

File: app/src/main/java/com/onboarding/wheatherapp/data/mapper/WeatherMapper.kt (hunk 12 above)
Comment (jnicolaslaurenti) [reply app/src/main/java/com/onboarding/wheatherapp/data/mapper/WeatherMapper.kt]: Solved!

**Review by lucianoeli — APPROVED** [review]

**Review by BQuinteros — APPROVED** [review]

**Review by mauroLonghi — APPROVED** [review]

**Pull request merged**